core/src/main/scala/org/apache/spark/util/ManualClock.scala

```diff
@@ -27,6 +27,7 @@ package org.apache.spark.util
 private[spark] class ManualClock(private var time: Long) extends Clock {
 
   private var _isWaiting = false
+  private var _readyForFirstPeek = false
```

Comment:
This sounds good!

One thing to confirm: we are assuming only **one** main thread and **one** test thread, right? Because:
- given two main threads m1, m2, and one testing thread t, should `isThreadWaitingAt(time)` return true or false if m1 reaches `waitUtil(time)` but m2 does not yet?
- given one main threads m, and two testing threads t1, t2, should `isThreadWaitingAt(time)` return true for only one of t1, t2 or both?